app/models/manager_refresh/inventory_collection_default.rb

```diff
@@ -1,184 +1,224 @@
+      :name     => "unknown",
+      :location => "unknown",
+      :template => true,
+    }.freeze,
+  }.freeze
+
+  HARDWARE_ATTRIBUTES = {
+    :model_class                  => ::Hardware,
+    :association                  => :hardwares,
+    :manager_ref                  => %i(vm_or_template).freeze,
+    # TODO(lsmola) just because of default value on cpu_sockets,
+    # this can be fixed by separating instances_hardwares and images_hardwares
+    :use_ar_object                => true,
+    :parent_inventory_collections => %i(vms miq_templates).freeze,
+
+    :inventory_object_attributes  => %i(
+      annotation
+      cpu_cores_per_socket
+      cpu_sockets
+      cpu_speed
+      cpu_total_cores
+      cpu_type
+      guest_os
+      manufacturer
+      memory_mb
+      model
+      networks
+      number_of_nics
+      serial_number
+      virtual_hw_version
+    ).freeze,
+
+    :targeted_arel                => lambda do |inventory_collection|
+      manager_uuids = inventory_collection.parent_inventory_collections.flat_map { |c| c.manager_uuids.to_a }
+      inventory_collection.parent.hardwares.joins(:vm_or_template).where('vms' => { :ems_ref => manager_uuids })
+    end.freeze,
+  }.freeze
+
+  OPERATING_SYSTEM_ATTRIBUTES = {
+    :model_class                  => ::OperatingSystem,
+    :association                  => :operating_systems,
+    :manager_ref                  => %i(vm_or_template).freeze,
+    :parent_inventory_collections => %i(vms miq_templates).freeze,
+
+    :inventory_object_attributes  => %i(
+      name
+      product_name
+      product_type
+      system_type
+      version
+    ).freeze,
+
+    :targeted_arel                => lambda do |inventory_collection|
+      manager_uuids = inventory_collection.parent_inventory_collections.flat_map { |c| c.manager_uuids.to_a }
+      inventory_collection.parent.operating_systems.joins(:vm_or_template)
+                          .where('vms' => { :ems_ref => manager_uuids })
+    end.freeze,
+  }.freeze
+
+  DISK_ATTRIBUTES = {
+    :model_class                  => ::Disk,
+    :association                  => :disks,
+    :manager_ref                  => %i(hardware device_name).freeze,
+    :parent_inventory_collections => %i(vms).freeze,
+
+    :inventory_object_attributes  => %i(
+      device_name
+      device_type
+      controller_type
+      present
+      filename
+      location
+      size
+      size_on_disk
+      disk_type
+      mode
+      bootable
+      storage
+    ).freeze,
+
+    :targeted_arel                => lambda do |inventory_collection|
+      manager_uuids = inventory_collection.parent_inventory_collections.flat_map { |c| c.manager_uuids.to_a }
+      inventory_collection.parent.disks.joins(:hardware => :vm_or_template)
+                          .where(:hardware => { 'vms' => { :ems_ref => manager_uuids } })
+    end.freeze,
+  }.freeze
+
+  class << self
+    private
+
+    def define_attribute_getters
```

Comment:
In this case, I don't mind the code not being 100% dry. It's better to be able to easily grep each config (or having IDE just click through, which we can do with then ansible version of the persister)

So less meta is better in this case :-)